File: level3/순위.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<vector<int>> results) {
    int answer = 0;
    int array[n+1][n+1];
    int count;
    
    for(int i=1; i<=n; i++)
        for(int j=1; j<=n; j++)
            array[i][j] = 0;

    for(int i=0; i<results.size(); i++){
        array[results[i][0]][results[i][1]] = 1;
        array[results[i][1]][results[i][0]] = -1;
    }

    for(int i=1; i<=n; i++)
        for(int j=1; j<=n; j++)
            if(array[i][j] == 1)
                for(int k=1; k<=n; k++)
                    if(array[j][k] == 1){
                        array[i][k] = 1;
                        array[k][i] = -1;
                    }

    for(int i=1; i<=n; i++)
        for(int j=1; j<=n; j++)
            if(array[i][j] == -1)
                for(int k=1; k<=n; k++)
                    if(array[j][k] == -1){
                        array[i][k] = -1;
                        array[k][i] = 1;
                    }

    for(int i=1; i<=n; i++){
        count = 0;
        for(int j=1; j<=n; j++){
            if(array[i][j] != 0)
                count++;
        }
        if(count == n-1)
            answer++;
    }
    return answer;
}

File: JavaScript/level3/순위.js
function solution(n, results) {
    var answer = 0;
    
    let result = new Array(n).fill(null).map((x) => new Array(n).fill(0))
    
    for(let x of results){
        result[x[0]-1][x[1]-1]=1;
        result[x[1]-1][x[0]-1]=-1;
    }
    
    for(let i=0; i<n; i++)
        for(let j=0; j<n; j++)
            if(result[i][j] === 1)
                for(let k=0; k<n; k++)
                    if(result[j][k] === 1){
                        result[i][k] = 1;
                        result[k][i] = -1;
                    }
                
    for(let i=0; i<n; i++)
        for(let j=0; j<n; j++)
            if(result[i][j] === -1)
                for(let k=0; k<n; k++)
                    if(result[j][k] === -1){
                        result[i][k] = -1;
                        result[k][i] = 1;
                    }
                
    for(let i=0; i<n; i++){
        let temp = result[i].filter(x => x != 0);
        if(temp.length === n-1)
            answer++;
    }
    
    return answer;
}